py3status/modules/battery_level.py

```diff
@@ -237,8 +251,69 @@ def _refresh_battery_info(self):
         acpi_list = [acpi_list[i:i + 2]
                      for i in range(0, len(acpi_list) - 1, 2)]
 
-        battery_list = [self._extract_battery_information_from_acpi(battery)
-                        for battery in acpi_list]
+        return [_parse_battery_info(battery) for battery in acpi_list]
+
+    def _extract_battery_information_from_sys(self):
+        """
+        Extract the percent charged, charging state, time remaining,
+        and capacity for a battery, using Linux's kernel /sys interface
+
+        Only available in kernel 2.6.24(?) and newer. Before kernel provided
+        a similar, yet incompatible interface in /proc
+        """
+        def _parse_battery_info(sys_path):
+            """
+            Extract battery information from uevent file, already convert to
+            int if necessary
+            """
+            raw_values = dict()
+            with open(sys_path + "/uevent") as f:
+                for var in f.read().split():
+                    k, v = var.split("=")
+                    try:
+                        raw_values[k] = int(v)
+                    except ValueError:
+                        raw_values[k] = v
+            return raw_values
+
+        battery_list = []
+        for path in iglob(os.path.join(self.sys_battery_path, "BAT*")):
+            r = _parse_battery_info(path)
+            battery = dict()
+            battery["capacity"] = r["POWER_SUPPLY_ENERGY_FULL"]
+            battery["charging"] = "Charging" in r["POWER_SUPPLY_STATUS"]
+            battery["percent_charged"] = math.floor(
+                r["POWER_SUPPLY_ENERGY_NOW"] / battery["capacity"] * 100)
```

Comment:
This returns a float eg `5.0` so we always have the `.0`  using `int()` instead of `math.floor()` would fix this